api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java

```diff
@@ -666,9 +683,169 @@ public void shortenedStackTrace_shouldReturnNullIfStackTraceIsNull() {
 	}
 	
 	/**
-	 * @see OpenmrsUtil#nullSafeEqualsIgnoreCase(String,String)
+	 * @see OpenmrsUtil#folderContains(File, String)
+	 */
+	@Test
+	public void folderContains_shouldReturnTrueWhenFolderContainsFileWithGivenName() {
+		File folder = createFolderInUserDirectory("folder");
+		String separator = FileSystems.getDefault().getSeparator();
+		File file = new File(folder.getPath()  + separator + "file");
+		try {
+			file.createNewFile();
+			assertTrue(OpenmrsUtil.folderContains(folder, "file"));
+			file.delete();
+
+		} 
+		catch (IOException e) {}
+		folder.delete();
+	}
+	
+	/**
+	 * @see OpenmrsUtil#folderContains(File, String)
+	 */
+	@Test
+	public void folderContains_shouldReturnFalseWhenFolderDoesNotContainFileWithGivenName() {
+		File folder = createFolderInUserDirectory("folder");
+		assertFalse(OpenmrsUtil.folderContains(folder,"file"));
+		folder.delete();
+	}
+	
+	/**
+	 * @see OpenmrsUtil#folderContains(File, String)
+	 */
+	@Test
+	public void folderContains_shouldReturnFalseWhenGivenStringIsNull() {
+		File folder = createFolderInUserDirectory("folder");
+		assertFalse(OpenmrsUtil.folderContains(folder, null));
+		folder.delete();
+	}
+	
+	/**
+	 * @see OpenmrsUtil#folderContains(File, String)
+	 */
+	@Test
+	public void folderContains_shouldReturnFalseWhenGivenFolderIsNull() {
+		assertFalse(OpenmrsUtil.folderContains(null,"file"));
+	}
+	
+	/**
+	 * @see OpenmrsUtil#conceptListHelper(String)
+	 */
+	@Test
+	public void conceptListHelper_shouldReturnEmptyListWhenGivenDescriptorIsNullOrEmpty() {
+		List<Concept> list = OpenmrsUtil.conceptListHelper("");
+		assertTrue(list.size() == 0);
+
+		list = OpenmrsUtil.conceptListHelper(null);
+		assertTrue(list.size() == 0);
+	}
+
+	/**
+	 * @see OpenmrsUtil#conceptListHelper(String)
+	 */
+	@Test
+	public void conceptListHelper_shouldReturnListOfConceptsWithGivenNameInDescriptor() {
+		List<Concept> helperList = OpenmrsUtil.conceptListHelper("name:YES|name:NO|name:DIED|set:25|set:26");
+		List<Concept> serviceList = new ArrayList<>();
+		List<Concept> aux;
+		Concept con;
+		//Adding the concepts in the list to compare with the first one
+		ConceptService cs = Context.getConceptService();
```

Comment:
I see alot of code duplication of the above line. Can we have it as a class variable?

Reply:
ok

Reply:
If i do that i get exception : no service found APIexception

Reply:
Make it a class level variable and call `Context.getConceptService();` from a method level. eg.

```java
@Before
public void setup() {
   // Initialization goes here
}
```